ui/pages/smart-transactions/smart-transaction-status-page/smart-transaction-status-page.tsx

```diff
@@ -0,0 +1,434 @@
+          setIsSmartTransactionTakingTooLong(true);
+        }
+        return;
+      }
+
+      setTimeLeftForPendingStxInSec(stxDeadline - secondsAfterStxSubmission);
+    };
+
+    const intervalId = setInterval(calculateRemainingTime, 1000);
+    calculateRemainingTime();
+
+    // eslint-disable-next-line consistent-return
+    return () => clearInterval(intervalId);
+  }, [
+    isSmartTransactionPending,
+    isSmartTransactionTakingTooLong,
+    smartTransaction?.creationTime,
+    stxDeadline,
+  ]);
+
+  return {
+    timeLeftForPendingStxInSec,
+    isSmartTransactionTakingTooLong,
+    stxDeadline,
+  };
+};
+
+export const SmartTransactionStatusPage = ({
+  requestState,
+  onCloseExtension,
+  onViewActivity,
+}: SmartTransactionStatusPageProps) => {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const { smartTransaction, isDapp } = requestState;
+  const isSmartTransactionPending =
+    !smartTransaction ||
+    smartTransaction.status === SmartTransactionStatuses.PENDING;
+  const isSmartTransactionSuccess =
+    smartTransaction?.status === SmartTransactionStatuses.SUCCESS;
+  const isSmartTransactionCancelled = Boolean(
+    smartTransaction?.status?.startsWith(SmartTransactionStatuses.CANCELLED),
+  );
+  const featureFlags: {
+    smartTransactions?: {
+      expectedDeadline?: number;
+      maxDeadline?: number;
+    };
+  } | null = useSelector(getFeatureFlagsByChainId);
+  const stxEstimatedDeadline =
+    featureFlags?.smartTransactions?.expectedDeadline ||
+    SMART_TRANSACTIONS_EXPECTED_DEADLINE_FALLBACK;
+  const stxMaxDeadline =
+    featureFlags?.smartTransactions?.maxDeadline ||
+    SMART_TRANSACTIONS_MAX_DEADLINE_FALLBACK;
+  const {
+    timeLeftForPendingStxInSec,
+    isSmartTransactionTakingTooLong,
+    stxDeadline,
+  } = useRemainingTime({
+    isSmartTransactionPending,
+    smartTransaction,
+    stxMaxDeadline,
+    stxEstimatedDeadline,
+  });
+  const chainId: string = useSelector(getCurrentChainId);
+
+  const countdown = isSmartTransactionPending ? (
+    <Text
+      display={Display.InlineBlock}
+      textAlign={TextAlign.Center}
+      color={TextColor.textAlternative}
+      variant={TextVariant.bodySm}
+      className="smart-transaction-status-page__countdown"
+    >
+      {showRemainingTimeInMinAndSec(timeLeftForPendingStxInSec)}
+    </Text>
+  ) : undefined;
+
+  const { title, description, iconName, iconColor } = getDisplayValues({
+    t,
+    countdown,
+    isSmartTransactionPending,
+    isSmartTransactionTakingTooLong,
+    isSmartTransactionSuccess,
+    isSmartTransactionCancelled,
+  });
+
+  useEffect(() => {
+    dispatch(hideLoadingIndication());
+  }, []);
+
+  const uuid = smartTransaction?.uuid;
+  const portfolioSmartTransactionStatusUrl =
+    uuid && chainId
+      ? `${BaseUrl.Portfolio}/networks/${Number(
+          hexToDecimal(chainId),
+        )}/smart-transactions/${uuid}`
+      : undefined;
+
+  const Deadline = () => {
```

Comment:
Minor, these mini-components are awesome, but they are still nested within the main component.

A benefit of defining them as stand-alone functions at the top level of this file is greater decoupling as they can use their own hooks such as `useState` and `useEffect` etc, and be passed their own properties.